packages/scene-composer/src/components/three-fiber/GroundPlane.tsx

```diff
@@ -0,0 +1,93 @@
+import * as THREE from 'three';
+import React, { useRef, useEffect, useState } from 'react';
+import { ThreeEvent, useFrame } from '@react-three/fiber';
+import { Plane } from '@react-three/drei';
+
+import { IGroundPlaneSettings, KnownSceneProperty } from '../../interfaces';
+import { useSceneComposerId } from '../../common/sceneComposerIdContext';
+import useAddWidget from '../../hooks/useAddWidget';
+import useMatterportViewer from '../../hooks/useMatterportViewer';
+import useTwinMakerTextureLoader from '../../hooks/useTwinMakerTextureLoader';
+import { useEditorState, useStore } from '../../store';
+import { acceleratedRaycasting } from '../../utils/objectThreeUtils';
+
+const GroundPlane: React.FC = () => {
+  const meshRef = useRef<THREE.Mesh>();
+  const sceneComposerId = useSceneComposerId();
+  const { isEditing, addingWidget } = useEditorState(sceneComposerId);
+  const { handleAddWidget } = useAddWidget();
+  const { loadTextureOnMesh, clearTextureOnMesh } = useTwinMakerTextureLoader();
+  const groundPlaneSettings = useStore(sceneComposerId)((state) =>
+    state.getSceneProperty<IGroundPlaneSettings>(KnownSceneProperty.GroundPlaneSettings),
+  );
+  const { enableMatterportViewer } = useMatterportViewer();
+
+  const [meshId, setMeshId] = useState('');
+  const [dirtyTexture, setDirtyTexture] = useState(false);
+
+  const MAX_CLICK_DISTANCE = 2;
```

Comment:
Is this a reusable value that should be sourced from a constants file or is it truly unique to this component?

Reply:
It's in several components at this point and should be moved to a global constants probably.